rpm/__init__.py

```diff
@@ -71,6 +71,18 @@ def get_sitepackages_and_suffixes(interpreter):
     for interpreter in interpreters:
         if not Path(interpreter).is_file():
             continue
+        # skip interpreters with a different major.minor version than the
+        # running interpreter
+        # importing their rpm module has most likely catastrophic consequences
```

Comment:
Would it make sense to reorder the interpreters above and try `python{majorver}.{minorver}` first instead?. Since https://github.com/rpm-software-management/rpm/issues/2345, RPM python bindings are built against the Python 3.7+ stable API, so this change could break things that worked before, but it also would prevent segfaults and other nastiness in cases where they aren't.

On Fedora 40 with a Python 3.12 system interpreter, this seems to work:

```
[gotmax@fedora] ~ >>> rpm -qR python3-rpm | grep python
python(abi) = 3.12
[gotmax@fedora] ~ >>> uvx --from rpm --python 3.7 python -c 'import rpm; print(rpm.expandMacro("%_bindir"))'
/usr/bin
```